packages/amplify-cli/src/utils/run-build-command.ts

```diff
@@ -0,0 +1,29 @@
+import { $TSContext, AmplifyError, spinner } from 'amplify-cli-core';
+import { printer } from 'amplify-prompts';
+import { sync as execaSync } from 'execa';
+
+export default async function runBuildCommand(context: $TSContext) {
+  if (
+    context.exeInfo.inputParams.runBuildCommand &&
+    context.exeInfo.projectConfig.frontend === 'javascript' &&
+    context.exeInfo.projectConfig?.javascript?.config?.BuildCommand
+  ) {
+    const { BuildCommand } = context.exeInfo.projectConfig.javascript.config;
+    const buildCommandArray = BuildCommand.split(' ');
+    const buildCommandBase = buildCommandArray[0];
+    const buildCommandArgs = buildCommandArray.slice(1);
+    try {
+      printer.info(`Executing build command: ${BuildCommand}`);
+      spinner.start('Building app...');
+      await execaSync(buildCommandBase, buildCommandArgs, {
+        stdio: 'inherit',
+      });
+      spinner.succeed('Successfully built app with build command.');
+    } catch (e) {
+      throw new AmplifyError('ConfigurationError', {
```

Comment:
pass the underlying error to the constructor as the 3rd arg. Also can remove the resolution unless there is additional useful context that can be provided here.